PyRDF/backend/Spark.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import print_function
+from .Dist import Dist
+from pyspark import SparkConf, SparkContext
+
+class Spark(Dist):
+    """
+    Backend that executes the computational graph
+    using using `Spark` framework for distributed
+    execution.
+
+    """
+    def __init__(self, config = {}):
+        """
+        Creates an instance of the Spark
+        backend class.
+
+        Parameters
+        ----------
+        config : dict (optional)
+            The config options for Spark backend. The default
+            value is an empty Python dictionary `{}`. Config
+            specifically reads three options, 'SparkContext'
+            or 'SparkConf' and 'npartitions'. If 'SparkContext'
+            is not given, 'SparkContext' uses 'SparkConf' option,
+            which by default is 'None' and 'npartitions' is
+            '2' by default.
+
+        """
+        super(Spark, self).__init__(config)
+
+        try:
+            # Remove the value of 'npartitions' from config dict
+            self.npartitions = config.pop('npartitions')
+        except KeyError:
+            # Set the value to 'None' if 'npartitions'
+            # doesn't exist
+            self.npartitions = None
+
+        sparkConf = SparkConf().setAll(config.items())
+        self.sparkContext = SparkContext.getOrCreate(sparkConf)
+
+        # Set the value of 'npartitions' if it doesn't exist
+        self.npartitions = self.npartitions or self.sparkContext.getConf().get('spark.executor.instances') or 2
```

Comment:
Indeed this is the right logic:
- First try with user config for npartitions
- Second try to get default/user configured from SparkContext
- Last option, just give it a value (2 is fine).